### PR TITLE
CircleCI 2.0 - WIP

### PR DESCRIPTION
**Pull request by DarthHater**
I rule and have completed some work on Case Manager that's ready for review!

This is far from done but very WIP rough PR of CircleCI 2.0 to start working through issues. YAML validator passes, so uh, YOLO!

This pull request makes the following changes:
* New `.circleci` folder
* New `config.yml` file

It relates to the following issue #s: 
* Fixes #953

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# Empty
+version: 2
+
+jobs:
+  build:
+    parallelism: 3
+    working_directory: ~/abortioneering
+    docker:
+      - image: circleci/ruby:2.4.3-node-browsers
```
Comment (colinxfleming) [inline .circleci/config.yml]: does this conform to the yaml spec? aren't arrays supposed to be aligned with their parent key? e.g.:

```
docker:
- image: circleci/ruby:2.4.3
  environment:
    WHATEV: true
- image: circleci/mongo:latest

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        environment:
+          BUNDLE_JOBS: 3
+          BUNDLE_RETRY: 3
+          BUNDLE_PATH: vendor/bundle
+          RAILS_ENV: test
+      - image: circleci/mongo:latest
+    steps:
+      - checkout
```
Comment (colinxfleming) [inline .circleci/config.yml]: same yaml alignment question here

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      - run: 
+          name: Which bundler?
+          command: bundle -v
+      - restore_cache:
+          keys:
+            - dcaf_case_management-{{ checksum "Gemfile.lock" }}
```
Comment (colinxfleming) [inline .circleci/config.yml]: yaml indent?

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+          key: dcaf_case_management-{{ checksum "Gemfile.lock" }}
+          paths:
+            - vendor/bundle
+      - run: bundle exec rake knapsack:minitest
+      - run: gem install --no-rdoc brakeman; bundle exec brakeman --exit-on-warn .
+      - run: gem install --no-rdoc ruby_audit; bundle exec ruby-audit check
+      - run: gem install --no-rdoc bundler-audit; bundle-audit update; bundle-audit check
+  deployment:
+    steps:
+      - run: yes | gem install --no-rdoc brakeman-pro --source $BRAKEMAN_PRO_URL; bundle exec brakeman --exit-on-warn .
```
Comment (colinxfleming) [inline .circleci/config.yml]: yaml????

File: .circleci/config.yml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+# general:
+  # artifacts:
+    # - "tmp/screenshots/"
+
+workflows:
+  version: 2
+  dcaf_workflow_build_test_deploy:
+    jobs:
+      - build
```
Comment (colinxfleming) [inline .circleci/config.yml]: el yaml

File: .circleci/config.yml [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+      - deployment:
+        filters:
+          branches:
+            only: master
+        requires:
+          - build
```
Comment (colinxfleming) [inline .circleci/config.yml]: luv 2 yaml. live 2 yaml

**Comment by colinxfleming** [comment]
I figured out how to troubleshoot the orange wrench. Looks like you can get to error messages via clicking your branch, and then clicking workflows. Here's the output:

```
We weren't able to start this workflow.
Encountered errors trying to create workflows from this config: Config does not conform to schema: {:workflows {:dcaf_workflow_build_test_deploy {:jobs [nil {:deployment (not (map? nil)), :filters {:branches disallowed-key}, :requires (not (map? a-clojure.lang.LazySeq))}]}}}
For more examples see the Workflows documentation.
```

Workflows documentation is here: https://circleci.com/docs/2.0/workflows/

**Comment by colinxfleming** [comment]
The deal wound up being needing to declare parts of the job as `docker`, `machine`, or `macos`. We declared everything as docker here. Test splitting is also working properly!

**Comment by DarthHater** [comment]
Picking this back up @colinxfleming , sorry for the lag.

File: .circleci/config.yml [hunk 7]
```diff
@@ -0,0 +1,77 @@
+# Empty
+version: 2
+
+jobs:
+  build:
+    parallelism: 4
+    working_directory: ~/abortioneering
```
Comment (DarthHater) [inline .circleci/config.yml]: lol

File: .circleci/config.yml [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+      - image: circleci/ruby:2.5.1-node-browsers
+    steps:
+      - run: git push git@heroku.com:dcaf-cmapp-staging.git $CIRCLE_SHA1:master
+      - run: docker build -t colinxfleming/dcaf_case_management -f .docker/Dockerfile . && docker login -e "$DOCKEREMAIL" -u "$DOCKERLOGIN" -p "$DOCKERPASS" && docker push colinxfleming/dcaf_case_management
+
+# general:
+  # artifacts:
+    # - "tmp/screenshots/"
+
+# test:
```
Comment (DarthHater) [inline .circleci/config.yml]: This section can be deleted

Comment (DarthHater) [reply .circleci/config.yml]: In: ef3b4cf

File: .circleci/config.yml [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+      - run: bundle exec rake assets:clobber assets:precompile
+      - run: "[[ ! -s \"$(git rev-parse --git-dir)/shallow\" ]] || git fetch --unshallow" # necessary due to moving to an organization
+  deploy:
+    docker:
+      - image: circleci/ruby:2.5.1-node-browsers
+    steps:
+      - run: git push git@heroku.com:dcaf-cmapp-staging.git $CIRCLE_SHA1:master
+      - run: docker build -t colinxfleming/dcaf_case_management -f .docker/Dockerfile . && docker login -e "$DOCKEREMAIL" -u "$DOCKERLOGIN" -p "$DOCKERPASS" && docker push colinxfleming/dcaf_case_management
+
+# general:
```
[the remark this answers is not in the document]
Comment (colinxfleming) [reply .circleci/config.yml]: I think systemtest does it out of the box now, but let's see.

**Comment by colinxfleming** [comment]
image

**Comment by colinxfleming** [comment]
image

**Comment by colinxfleming** [comment]
ew does github not do animated gifs in comments? SICK

**Pull request merged**